lib/statsd.js

```diff
@@ -38,20 +51,42 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
     let socket;
     let errMessage;
 
-    if (args.protocol === 'tcp') {
+    if (args.protocol === TCP) {
       try {
         socket = net.connect(args.port, args.host);
         socket.setKeepAlive(true);
-      } catch (e) {
-        errMessage = `Could not establish connection to ${args.host}:${args.port}`;
-        if (instance.errorHandler) {
```

Comment:
Moved the errorHandler bit out of the if block- it's still in place for use below